### PR TITLE
format with gofumpt

### PR DESCRIPTION
**Pull request by 6543**
as title

### CODE REVIEW

File: services/migrations/gogs_test.go [hunk 1]
```diff
@@ -99,8 +99,8 @@ func TestGogsDownloadRepo(t *testing.T) {
 			Content:     "test",
 			Milestone:   "",
 			State:       "open",
-			Created:     time.Date(2019, 06, 11, 8, 16, 44, 0, time.UTC),
-			Updated:     time.Date(2019, 10, 26, 11, 07, 2, 0, time.UTC),
+			Created:     time.Date(2019, 0o6, 11, 8, 16, 44, 0, time.UTC),
```
Comment (6543) [inline services/migrations/gogs_test.go]: this could be bugs!

File: services/migrations/gitlab_test.go [hunk 2]
```diff
@@ -327,7 +329,7 @@ func TestGitlabDownloadRepo(t *testing.T) {
 		{
 			ReviewerID:   4575606,
 			ReviewerName: "real6543",
-			CreatedAt:    time.Date(2020, 04, 19, 19, 24, 21, 108000000, time.UTC),
+			CreatedAt:    time.Date(2020, 0o4, 19, 19, 24, 21, 108000000, time.UTC),
```
Comment (6543) [inline services/migrations/gitlab_test.go]: bug?

File: services/migrations/github_test.go [hunk 3]
```diff
@@ -364,7 +364,7 @@ func TestGitHubDownloadRepo(t *testing.T) {
 			ReviewerID:   165205,
 			ReviewerName: "lafriks",
 			CommitID:     "076160cf0b039f13e5eff19619932d181269414b",
-			CreatedAt:    time.Date(2019, 11, 12, 21, 38, 00, 0, time.UTC),
+			CreatedAt:    time.Date(2019, 11, 12, 21, 38, 0o0, 0, time.UTC),
```
Comment (6543) [inline services/migrations/github_test.go]: same

File: services/mailer/mailer_test.go [hunk 4]
```diff
@@ -14,14 +14,14 @@ import (
...
-	var mailService = setting.Mailer{
+	mailService := setting.Mailer{
 		From: "test@gitea.com",
 	}
 
 	setting.MailService = &mailService
 	setting.Domain = "localhost"
 
-	date := time.Date(2000, 01, 02, 03, 04, 05, 06, time.UTC)
+	date := time.Date(2000, 0o1, 0o2, 0o3, 0o4, 0o5, 0o6, time.UTC)
```
Comment (6543) [inline services/mailer/mailer_test.go]:
```suggestion
	date := time.Date(2000, 1, 2, 3, 4, 5, 6, time.UTC)
```

File: services/migrations/gogs_test.go (hunk 1 above)
Comment (Gusted) [reply services/migrations/gogs_test.go]: Yup that's a bug in gofumpt, at least it's not expected to have octal here
.

Comment (jolheiser) [reply services/migrations/gogs_test.go]: https://github.com/mvdan/gofumpt#added-rules

> Octal integer literals should use the 0o prefix on modules using Go 1.13 and later
example
const perm = 0755
const perm = 0o755

File: routers/web/web.go [hunk 5]
```diff
@@ -189,7 +189,7 @@ func RegisterRoutes(m *web.Route) {
 	ignSignInAndCsrf := context.Toggle(&context.ToggleOptions{DisableCSRF: true})
 	reqSignOut := context.Toggle(&context.ToggleOptions{SignOutRequired: true})
 
-	//bindIgnErr := binding.BindIgnErr
+	// bindIgnErr := binding.BindIgnErr
```
Comment (6543) [inline routers/web/web.go]:
```suggestion
```

File: services/migrations/gogs_test.go (hunk 1 above)
Comment (Gusted) [reply services/migrations/gogs_test.go]: Yeah, but they should IMO only be used on such integers, and not with other non-octal digits, as this makes this argument inconsistent with the others.

File: services/migrations/gogs_test.go [hunk 6]
```diff
@@ -121,8 +121,8 @@ func TestGogsDownloadRepo(t *testing.T) {
 			PosterID:    5331,
 			PosterName:  "lunny",
 			PosterEmail: "xiaolunwen@gmail.com",
-			Created:     time.Date(2019, 06, 11, 8, 19, 50, 0, time.UTC),
-			Updated:     time.Date(2019, 06, 11, 8, 19, 50, 0, time.UTC),
+			Created:     time.Date(2019, 0o6, 11, 8, 19, 50, 0, time.UTC),
+			Updated:     time.Date(2019, 0o6, 11, 8, 19, 50, 0, time.UTC),
```
Comment (6543) [inline services/migrations/gogs_test.go]:
```suggestion
			Created:     time.Date(2019, 6, 11, 8, 19, 50, 0, time.UTC),
			Updated:     time.Date(2019, 6, 11, 8, 19, 50, 0, time.UTC),
```

File: services/migrations/gogs_test.go [hunk 7]
```diff
@@ -99,8 +99,8 @@ func TestGogsDownloadRepo(t *testing.T) {
 			Content:     "test",
 			Milestone:   "",
 			State:       "open",
-			Created:     time.Date(2019, 06, 11, 8, 16, 44, 0, time.UTC),
-			Updated:     time.Date(2019, 10, 26, 11, 07, 2, 0, time.UTC),
+			Created:     time.Date(2019, 0o6, 11, 8, 16, 44, 0, time.UTC),
+			Updated:     time.Date(2019, 10, 26, 11, 0o7, 2, 0, time.UTC),
```
Comment (6543) [inline services/migrations/gogs_test.go]:
```suggestion
			Created:     time.Date(2019, 6, 11, 8, 16, 44, 0, time.UTC),
			Updated:     time.Date(2019, 10, 26, 11, 7, 2, 0, time.UTC),
```

File: services/migrations/gitlab_test.go (hunk 2 above)
Comment (6543) [inline services/migrations/gitlab_test.go]:
```suggestion
			CreatedAt:    time.Date(2020, 4, 19, 19, 24, 21, 108000000, time.UTC),
```

File: services/migrations/gitlab_test.go [hunk 8]
```diff
@@ -310,13 +312,13 @@ func TestGitlabDownloadRepo(t *testing.T) {
...
 			ReviewerName: "zeripath",
-			CreatedAt:    time.Date(2019, 11, 28, 16, 02, 8, 377000000, time.UTC),
+			CreatedAt:    time.Date(2019, 11, 28, 16, 0o2, 8, 377000000, time.UTC),
 			State:        "APPROVED",
 		},
 		{
 			ReviewerID:   527793,
 			ReviewerName: "axifive",
-			CreatedAt:    time.Date(2019, 11, 28, 16, 02, 8, 377000000, time.UTC),
+			CreatedAt:    time.Date(2019, 11, 28, 16, 0o2, 8, 377000000, time.UTC),
```
Comment (6543) [inline services/migrations/gitlab_test.go]:
```suggestion
			CreatedAt:    time.Date(2019, 11, 28, 16, 2, 8, 377000000, time.UTC),
```

File: services/migrations/gitlab_test.go [hunk 9]
```diff
@@ -310,13 +312,13 @@ func TestGitlabDownloadRepo(t *testing.T) {
 		{
 			ReviewerID:   4102996,
 			ReviewerName: "zeripath",
-			CreatedAt:    time.Date(2019, 11, 28, 16, 02, 8, 377000000, time.UTC),
+			CreatedAt:    time.Date(2019, 11, 28, 16, 0o2, 8, 377000000, time.UTC),
```
Comment (6543) [inline services/migrations/gitlab_test.go]:
```suggestion
			CreatedAt:    time.Date(2019, 11, 28, 16, 2, 8, 377000000, time.UTC),
```

File: services/migrations/github_test.go [hunk 10]
```diff
@@ -421,8 +421,8 @@ func TestGitHubDownloadRepo(t *testing.T) {
 					Position:  4,
 					CommitID:  "2be9101c543658591222acbee3eb799edfc3853d",
 					PosterID:  81045,
-					CreatedAt: time.Date(2020, 01, 04, 11, 21, 41, 0, time.UTC),
-					UpdatedAt: time.Date(2020, 01, 04, 11, 21, 41, 0, time.UTC),
+					CreatedAt: time.Date(2020, 0o1, 0o4, 11, 21, 41, 0, time.UTC),
+					UpdatedAt: time.Date(2020, 0o1, 0o4, 11, 21, 41, 0, time.UTC),
```
Comment (6543) [inline services/migrations/github_test.go]:
```suggestion
					CreatedAt: time.Date(2020, 1, 4, 11, 21, 41, 0, time.UTC),
					UpdatedAt: time.Date(2020, 1, 4, 11, 21, 41, 0, time.UTC),
```

File: services/migrations/github_test.go [hunk 11]
```diff
@@ -410,7 +410,7 @@ func TestGitHubDownloadRepo(t *testing.T) {
 			ReviewerID:   81045,
 			ReviewerName: "lunny",
 			CommitID:     "2be9101c543658591222acbee3eb799edfc3853d",
-			CreatedAt:    time.Date(2020, 01, 04, 11, 21, 41, 0, time.UTC),
+			CreatedAt:    time.Date(2020, 0o1, 0o4, 11, 21, 41, 0, time.UTC),
```
Comment (6543) [inline services/migrations/github_test.go]:
```suggestion
			CreatedAt:    time.Date(2020, 1, 4, 11, 21, 41, 0, time.UTC),
```

File: services/migrations/github_test.go [hunk 12]
```diff
@@ -400,7 +400,7 @@ func TestGitHubDownloadRepo(t *testing.T) {
 			ReviewerID:   81045,
 			ReviewerName: "lunny",
 			CommitID:     "2be9101c543658591222acbee3eb799edfc3853d",
-			CreatedAt:    time.Date(2020, 01, 04, 06, 07, 06, 0, time.UTC),
+			CreatedAt:    time.Date(2020, 0o1, 0o4, 0o6, 0o7, 0o6, 0, time.UTC),
```
Comment (6543) [inline services/migrations/github_test.go]:
```suggestion
			CreatedAt:    time.Date(2020, 1, 4, 6, 7, 6, 0, time.UTC),
```

File: services/migrations/github_test.go [hunk 13]
```diff
@@ -389,8 +389,8 @@ func TestGitHubDownloadRepo(t *testing.T) {
 					Position:  3,
 					CommitID:  "2be9101c543658591222acbee3eb799edfc3853d",
 					PosterID:  81045,
-					CreatedAt: time.Date(2020, 01, 04, 05, 33, 06, 0, time.UTC),
-					UpdatedAt: time.Date(2020, 01, 04, 05, 33, 18, 0, time.UTC),
+					CreatedAt: time.Date(2020, 0o1, 0o4, 0o5, 33, 0o6, 0, time.UTC),
+					UpdatedAt: time.Date(2020, 0o1, 0o4, 0o5, 33, 18, 0, time.UTC),
```
Comment (6543) [inline services/migrations/github_test.go]:
```suggestion
					CreatedAt: time.Date(2020, 1, 4, 5, 33, 6, 0, time.UTC),
					UpdatedAt: time.Date(2020, 1, 4, 5, 33, 18, 0, time.UTC),
```

File: services/migrations/github_test.go [hunk 14]
```diff
@@ -378,7 +378,7 @@ func TestGitHubDownloadRepo(t *testing.T) {
 			ReviewerID:   81045,
 			ReviewerName: "lunny",
 			CommitID:     "2be9101c543658591222acbee3eb799edfc3853d",
-			CreatedAt:    time.Date(2020, 01, 04, 05, 33, 18, 0, time.UTC),
+			CreatedAt:    time.Date(2020, 0o1, 0o4, 0o5, 33, 18, 0, time.UTC),
```
Comment (6543) [inline services/migrations/github_test.go]:
```suggestion
			CreatedAt:    time.Date(2020, 1, 4, 5, 33, 18, 0, time.UTC),
```

File: services/migrations/github_test.go (hunk 3 above)
Comment (6543) [inline services/migrations/github_test.go]:
```suggestion
			CreatedAt:    time.Date(2019, 11, 12, 21, 38, 0, 0, time.UTC),
```

File: services/migrations/gogs_test.go (hunk 1 above)
Comment (6543) [reply services/migrations/gogs_test.go]: I changed them to be just normal integer ...

File: integrations/api_user_orgs_test.go [hunk 15]
```diff
@@ -54,7 +54,7 @@ func TestUserOrgs(t *testing.T) {
 }
 
 func getUserOrgs(t *testing.T, userDoer, userCheck string) (orgs []*api.Organization) {
-	var token = ""
+	token := ""
 	session := emptyTestSession(t)
 	if len(userDoer) != 0 {
 		session = loginUser(t, userDoer)
```
Comment (Gusted) [inline integrations/api_user_orgs_test.go:60]:
```suggestion
	session := emptyTestSession(t)
	if len(userDoer) != 0 {
	        var token string
		session = loginUser(t, userDoer)
```

File: integrations/git_smart_http_test.go [hunk 16]
```diff
@@ -18,7 +18,7 @@ func TestGitSmartHTTP(t *testing.T) {
 }
 
 func testGitSmartHTTP(t *testing.T, u *url.URL) {
-	var kases = []struct {
+	kases := []struct {
```
Comment (Gusted) [inline integrations/git_smart_http_test.go:21]: Typo 😄

File: modules/context/api_test.go [hunk 17]
```diff
@@ -16,7 +16,7 @@ import (
 
 func TestGenAPILinks(t *testing.T) {
 	setting.AppURL = "http://localhost:3000/"
-	var kases = map[string][]string{
+	kases := map[string][]string{
```
Comment (Gusted) [inline modules/context/api_test.go:19]: Typo

File: modules/git/submodule_test.go [hunk 18]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 func TestGetRefURL(t *testing.T) {
-	var kases = []struct {
+	kases := []struct {
```
Comment (Gusted) [inline modules/git/submodule_test.go:14]: Again... typo

File: modules/log/file.go [hunk 19]
```diff
@@ -76,7 +76,7 @@ func (mw *MuxWriter) SetFd(fd *os.File) {
 func NewFileLogger() LoggerProvider {
 	log := &FileLogger{
 		Filename:         "",
-		Maxsize:          1 << 28, //256 MB
+		Maxsize:          1 << 28, // 256 MB
```
Comment (Gusted) [inline modules/log/file.go:79]: Just off-topic, but now I see this anyway 1 << 28 is 268MB. Maybe `256e6` should be used, which is 256MB.

File: modules/markup/csv/csv_test.go [hunk 20]
```diff
@@ -15,7 +15,7 @@ import (
 
 func TestRenderCSV(t *testing.T) {
 	var render Renderer
-	var kases = map[string]string{
+	kases := map[string]string{
```
Comment (Gusted) [inline modules/markup/csv/csv_test.go:18]: Someone lost their `c` key?

File: modules/public/public_test.go [hunk 21]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 func TestParseAcceptEncoding(t *testing.T) {
-	var kases = []struct {
+	kases := []struct {
```
Comment (Gusted) [inline modules/public/public_test.go:14]: ... I will personally send a new `c` key,

File: routers/web/admin/admin_test.go [hunk 22]
```diff
@@ -11,7 +11,7 @@ import (
 )
 
 func TestShadowPassword(t *testing.T) {
-	var kases = []struct {
+	kases := []struct {
```
Comment (Gusted) [inline routers/web/admin/admin_test.go:14]: ...

File: routers/web/repo/editor_test.go [hunk 23]
```diff
@@ -17,7 +17,7 @@ import (
 func TestCleanUploadName(t *testing.T) {
 	unittest.PrepareTestEnv(t)
 
-	var kases = map[string]string{
+	kases := map[string]string{
```
Comment (Gusted) [inline routers/web/repo/editor_test.go:20]: 🎉

File: routers/web/repo/issue_test.go [hunk 24]
```diff
@@ -13,7 +13,7 @@ import (
 )
 
 func TestCombineLabelComments(t *testing.T) {
-	var kases = []struct {
+	kases := []struct {
```
Comment (Gusted) [inline routers/web/repo/issue_test.go:16]: Am I getting insane?

File: services/webhook/deliver_test.go [hunk 25]
```diff
@@ -19,7 +19,7 @@ func TestWebhookProxy(t *testing.T) {
 	setting.Webhook.ProxyURLFixed, _ = url.Parse(setting.Webhook.ProxyURL)
 	setting.Webhook.ProxyHosts = []string{"*.discordapp.com", "discordapp.com"}
 
-	var kases = map[string]string{
+	kases := map[string]string{
```
Comment (Gusted) [inline services/webhook/deliver_test.go:22]: :tad

**Review by Gusted — COMMENTED** [review]
Just a lot of off-topic not related things.

**Comment by 6543** [comment]
@Gusted I already have one off-topic in this pull - I like to have your suggestions in an extra pull - this diff is already huge!!!

**Review by Gusted — APPROVED** [review]
Voila

**Review by wxiaoguang — APPROVED** [review]
gofumpt really does a good job

**Comment by wxiaoguang** [comment]
ps: @Gusted IMO the `kase` is not a typo but choose by purpose, to bypass the keyword `case`, just like many Java developers like to write `klass` for a variable name to bypass the `class` keyword.

**Comment by Gusted** [comment]
> ps: @Gusted IMO the `kase` is not a typo but choose by purpose, to bypass the keyword `case`, just like many Java developers like to write `klass` for a variable name to bypass the `class` keyword.

Ah I see, thanks! It was so confusing to see it being used.

**Comment by wxiaoguang** [comment]
One more thing, this PR could replace an old PR 

* https://github.com/go-gitea/gitea/pull/15057

**Comment by 6543** [comment]
I'll merge once we have an release branch

**Comment by wxiaoguang** [comment]
> I'll merge once we have an release branch

I prefer to get it merged in 1.16. 

Reasons:
1. This is only a refactoring, doesn't change the logic.
2. Merging it now will avoid potential conflicts in future.
3. There are still some open PRs for 1.16 and it still gets fixes.

**Review by zeripath — DISMISSED** [review]
you need to update:

make fmt
make fmt-check

We need gofumpt to be autodownloaded too. 

And we probably also need to update the Dev documentation to explain you need gofumpt in vscode goland with explanations on how to enable that.

**Comment by wxiaoguang** [comment]
Conflicts come again .... I was thinking about we should get this merged as early as possible in 1.16 .....

**Comment by zeripath** [comment]
Agreed - but we need the above makefile things to be fixed.

Now fortunately I think just rerunning gofumpt on the code will make them go away.

(That of course will not necessarily work for conflicts that merging this PR will cause with other PRs.)

**Comment by wxiaoguang** [comment]
@6543 @zeripath 

I think we still need  `@$(GO) run build/code-batch-process.go {something} -s -w '{file-list}'` (code.gitea.io/gitea/build/codeformat)

It can group  the imports.

The output o f gofumpt could be: 

```go
import (
	"os"
	"path"

	"code.gitea.io/gitea/modules/log"  // gofumpt doesn't group imports

	"code.gitea.io/gitea/modules/httpcache"
	"code.gitea.io/gitea/modules/setting"
)
```

Our `code.gitea.io/gitea/build/codeformat` can  format the code above to:

```go
import (
	"os"
	"path"

	"code.gitea.io/gitea/modules/httpcache"
	"code.gitea.io/gitea/modules/log"
	"code.gitea.io/gitea/modules/setting"
)
```

**Comment by lunny** [comment]
> @6543 @zeripath
> 
> I think we still need `@$(GO) run build/code-batch-process.go {something} -s -w '{file-list}'` (code.gitea.io/gitea/build/codeformat)
> 
> It can group the imports.
> 
> The output o f gofumpt could be:
> 
> ```go
> import (
> 	"os"
> 	"path"
> 
> 	"code.gitea.io/gitea/modules/log"  // gofumpt doesn't group imports
> 
> 	"code.gitea.io/gitea/modules/httpcache"
> 	"code.gitea.io/gitea/modules/setting"
> )
> ```
> 
> Our `code.gitea.io/gitea/build/codeformat` can format the code above to:
> 
> ```go
> import (
> 	"os"
> 	"path"
> 
> 	"code.gitea.io/gitea/modules/httpcache"
> 	"code.gitea.io/gitea/modules/log"
> 	"code.gitea.io/gitea/modules/setting"
> )
> ```

I think so.

**Comment by 6543** [comment]
ok will do a follow up just for the `make fmt`

**Comment by 6543** [comment]
-> #18526

**Pull request merged**